### PR TITLE
setup CI with docker hub.

### PR DESCRIPTION
**Pull request by Diablo2050**
This has two CI files one for pull requests with tests only and the other has version tags and master push.

### CODE REVIEW

File: docker_push.sh [hunk 1]
```diff
@@ -5,25 +5,21 @@
...
-if [[ -z "$TRAVIS_PULL_REQUEST_BRANCH" ]]; then
-    if [[ -n "$TRAVIS_TAG" ]]; then
+if [[ -z "$GITHUB_BASE_REF" ]]; then
+    if [[ "$RELEASE_VERSION" == v*  ]]; then
         # Tagging should trigger a release in Docker Hub, that's immutable.
         echo "git tag action, push the tag"
-        docker_push "$TRAVIS_TAG-release"
-    elif [[ "$TRAVIS_BRANCH" == "master" ]]; then
+        docker_push "$RELEASE_VERSION-release"
+    elif [[ "$RELEASE_VERSION" == "dev" ]]; then
```
Comment (simplesteph) [inline docker_push.sh:16]: @Diablo2050 what's that line for?

File: docker_push.sh [hunk 2]
```diff
@@ -5,25 +5,21 @@
...
         # we push to latest when master is built and that's not a pull request
         echo "master build, push to latest"
         docker_push "latest"
-    elif [[ "$TRAVIS_BRANCH" == "v"* ]]; then
-        # this is a version branch and we push it.
-        echo "branch build, push to branch-latest"
-        docker_push "$TRAVIS_BRANCH-latest"
+    # elif [[ "$RELEASE_VERSION" == v* ]]; then
```
Comment (simplesteph) [inline docker_push.sh:20]: if comments, please remove?

**Review by simplesteph — CHANGES_REQUESTED** [review]
I also think there's a problem: it doesn't run the test suit on pushes to master, right? Any push to master also needs to be tested to ensure we don't push a bad build out there

File: docker_push.sh (hunk 1 above)
Comment (Diablo2050) [reply docker_push.sh]: will be removed, should be == "master"

File: docker_push.sh (hunk 2 above)
Comment (Diablo2050) [reply docker_push.sh]: will be removed.

**Comment by Diablo2050** [comment]
@simplesteph changes done and test passed.

**Pull request merged**